cmd/occollector/app/builder/processor_builder.go

```diff
@@ -0,0 +1,146 @@
+
+package builder
+
+import (
+	"time"
+
+	"github.com/spf13/viper"
+)
+
+// SenderType indicates the type of sender
+type SenderType string
+
+const (
+	// ThriftTChannelSenderType represents a thrift-format tchannel-transport sender
+	ThriftTChannelSenderType SenderType = "thrift-tchannel"
+	// ThriftHTTPSenderType represents a thrift-format http-transport sender
+	ThriftHTTPSenderType = "thrift-http"
+	// InvalidSenderType represents an invalid sender
+	InvalidSenderType = "invalid"
+)
+
+// JaegerThriftTChannelSenderCfg holds configuration for Jaeger Thrift Tchannel sender
+type JaegerThriftTChannelSenderCfg struct {
+	CollectorHostPorts        []string      `mapstructure:"collector-host-ports"`
+	DiscoveryMinPeers         int           `mapstructure:"discovery-min-peers"`
+	DiscoveryConnCheckTimeout time.Duration `mapstructure:"discovery-conn-check-timeout"`
+}
+
+// NewJaegerThriftTChannelSenderCfg returns an instance of JaegerThriftTChannelSenderCfg with default values
+func NewJaegerThriftTChannelSenderCfg() *JaegerThriftTChannelSenderCfg {
+	opts := &JaegerThriftTChannelSenderCfg{
+		DiscoveryMinPeers:         3,
+		DiscoveryConnCheckTimeout: 250 * time.Millisecond,
+	}
+	return opts
+}
+
+// JaegerThriftHTTPSenderCfg holds configuration for Jaeger Thrift HTTP sender
+type JaegerThriftHTTPSenderCfg struct {
+	CollectorEndpoint string            `mapstructure:"collector-endpoint"`
+	Timeout           time.Duration     `mapstructure:"timeout"`
+	Headers           map[string]string `mapstructure:"headers"`
+}
+
+// NewJaegerThriftHTTPSenderCfg returns an instance of JaegerThriftHTTPSenderCfg with default values
+func NewJaegerThriftHTTPSenderCfg() *JaegerThriftHTTPSenderCfg {
+	opts := &JaegerThriftHTTPSenderCfg{
+		Timeout: 5 * time.Second,
+	}
+	return opts
+}
+
+// QueuedSpanProcessorCfg holds configuration for the queued span processor
+type QueuedSpanProcessorCfg struct {
+	// Name is the friendly name of the processor
+	Name string
+	// NumWorkers is the number of queue workers that dequeue batches and send them out
+	NumWorkers int `mapstructure:"num-workers"`
+	// QueueSize is the maximum number of batches allowed in queue at a given time
+	QueueSize int `mapstructure:"queue-size"`
+	// Retry indicates whether queue processor should retry span batches in case of processing failure
+	RetryOnFailure bool `mapstructure:"retry-on-failure"`
+	// BackoffDelay is the amount of time a worker waits after a failed send before retrying
+	BackoffDelay time.Duration `mapstructure:"backoff-delay"`
+	// SenderType indicates the type of sender to instantiate
+	SenderType   SenderType `mapstructure:"sender-type"`
+	SenderConfig interface{}
+}
+
+// NewDefaultQueuedSpanProcessorCfg returns an instance of QueuedSpanProcessorCfg with default values
+func NewDefaultQueuedSpanProcessorCfg() *QueuedSpanProcessorCfg {
+	opts := &QueuedSpanProcessorCfg{
+		Name:           "default-queued-jaeger-sender",
+		NumWorkers:     10,
+		QueueSize:      5000,
```

Comment:
nit: maybe add a note about these default values in README.

Reply:
Good suggestion @songy23, added.